orchestrator/types.py

```diff
@@ -202,19 +208,14 @@ def is_optional_type(t: Any, test_type: Optional[type] = None) -> bool:
     False
     >>> is_optional_type(Optional[State], State)
     True
-    >>> is_optional_type(int)
-    False
     """
-    if get_origin(t):
-        if get_origin(t) == Union and None.__class__ in get_args(t):  # type:ignore
-            for arg in get_args(t):
-                if arg is None.__class__:
-                    continue
+    union_types = [Union, UnionType] if UnionType else [Union]
+    if get_origin(t) and get_origin(t) in union_types and None.__class__ in get_args(t):
```

Comment:
wondering if this can't be simplified to:
`
if get_origin(t) in union_types and None.__class__ in get_args(t):`

If get_origin(t) is None< I guess it won't be in union_types